src/main/java/com/lorepo/icplayer/client/module/text/TextPresenter.java

```diff
@@ -511,6 +513,7 @@ public void reset(boolean isOnlyWrongAnswers) {
 
 		for (int i = 0; i < view.getChildrenCount(); i++) {
 			TextElementDisplay child = view.getChild(i);
+            child.setResetStatus(true);
```

Comment:
Ten fragment kodu nie dawał mi spokoju po naszej rozmowie. Jestem obecnie na Twej gałęzi i przeniosłem child.setResetStatus(true); do reset() w GapWidget i działa.
```
@Override
	public void reset() {
		setText("");
		gapInfo.setResetStatus(true);
		this.setWorkMode();
	}
``` 
Zrobiłem to z dwóch powodów:
* intuicyjne jest to, że w resecie zmienia się status czy wykonał się reset - zwiększenie czytelności i zrozumienia kodu
* w więcej niż w tym fragmencie kodu wykonywana jest operacja reset na child. Przykładem jest `handleGradualHideAnswers`. Więc, przy każdym reset musiałbyś wykonać metodę setResetStatus aby nie powodować niejednoznaczności w stanie wasReset dla obecnego kodu.

Więc należy dodać moją propozycję lub wykonać setResetStatus w reszcie powiązanego kodu.

Testów nie puszczałem, więc prosiłbym abyś wprowadził tą zmianę a następnie przetestował bo mogłem coś przegapić/źle zrozumieć.